ktor-client/ktor-client-jetty/jvm/src/io/ktor/client/engine/jetty/JettyResponseListener.kt

```diff
@@ -75,7 +80,9 @@ internal class JettyResponseListener(
             headersBuilder.append(field.name, field.value)
         }
 
-        if (frame.isEndStream) backendChannel.close()
+        if (frame.isEndStream || request.method == HttpMethod.Head) {
```

Comment:
Shouldn't head response always have `isEndStream` set? Since a head response has no body, a headers frame is alwasys the last with end stream mark set, isn't it?

Reply:
Yes, it always the last, and no, there is no flag in it(at least in jetty). It was a bug discovered by the statement.